app/extensions/__init__.py

```diff
@@ -470,6 +471,38 @@ def current_user_has_edit_permission(self):
         rule = ObjectActionRule(obj=self, action=AccessOperation.WRITE)
         return rule.check()
 
+    # will grab edm representation of this object
+    # cache allows minimal calls to edm for same object, but has the potential
+    #   to return stale data
+    def get_edm_complete_data(self, use_cache=True):
+        from app.extensions import is_extension_enabled
+        from flask import current_app
+        import time
+
+        # this will prevent HoustonModel objects from using this
+        if FeatherModel not in self.__class__.__bases__:
+            raise NotImplementedError('only available on FeatherModels')
+        if not is_extension_enabled('edm'):
+            return None
+        # going to give cache a life of 5 min kinda arbitrarily
+        time_now = int(time.time())
+        if (
+            use_cache
+            and hasattr(self, '_edm_cached_data')
+            and time_now - self._edm_cached_data.get('_edm_cache_created', 0) < 300
```

Comment:
I really dislike arbitrary numbers in code. Can you give this 300 a name please?
One of the guys I mentored sent me a message this morning saying he'd thought of me when creating a SecondsPerMinute constant.